labgrid/driver/mqtt.py

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env python3
+
+import threading
+import time
+
+import attr
+
+from .common import Driver
+from ..factory import target_factory
+from ..protocol import PowerProtocol
+from ..step import step
+from ..util import Timeout
+
+
+class MQTTError(Exception):
+    pass
+
+@target_factory.reg_driver
+@attr.s(eq=False)
+class TasmotaPowerDriver(Driver, PowerProtocol):
+    bindings = {
+            "power": {"TasmotaPowerPort"}
+    }
+    delay = attr.ib(default=2.0, validator=attr.validators.instance_of(float))
+    _client = attr.ib(default=None)
+    _status_lock = attr.ib(default=threading.Lock())
+    _status = attr.ib(default=None)
+
+    def __attrs_post_init__(self):
+        super().__attrs_post_init__()
+        import paho.mqtt.client as mqtt
+        self._client = mqtt.Client()
+
+    def on_activate(self):
+        self._client.on_message = self._on_message
+        self._client.on_connect = self._on_connect
+        self._client.connect(self.power.host)
+        self._client.loop_start()
+
+    def on_deactivate(self):
+        self._client.loop_stop()
+
+    def _on_message(self, client, userdata, msg):
+        if msg.payload == b'ON':
+            status = True
+        elif msg.payload == b'OFF':
+            status = False
+        with self._status_lock:
```

Comment:
We don't need a `threading.Lock` for single python assignements or checks, so we can drop the lock from the driver here.